docs/design/2026-03-12_http-adhoc-toolset.md

```
@@ -0,0 +1,327 @@
+```json
+{
+  "tool_call_id": "call_xyz",
+  "approved": true
+}
+```
+
+That's it. The server derives what to save from the tool call's params (`suggested_endpoint`). No `save_prefixes`, no `remember` flag, no `save_session_data`. The approval itself is the signal to remember.
+
+**Session persistence (server path):** When `process_tool_decisions()` executes an approved HTTP ad-hoc tool call, it stores the `suggested_endpoint` in tool result message metadata:
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)